pulsar-functions/api-java/src/main/java/org/apache/pulsar/functions/api/Context.java

```diff
@@ -161,15 +161,19 @@
 
     /**
      * Publish an object using serDe for serializing to the topic
-     * @param topicName The name of the topic for publishing
-     * @param object The object that needs to be published
-     * @param serDeClassName The class name of the class that needs to be used to serialize the object before publishing
+     *
+     * @param topicName
+     *            The name of the topic for publishing
+     * @param object
+     *            The object that needs to be published
+     * @param topicsPattern
+     *            Either a builtin schema type (eg: "avro", "json", "protobuf") or the class name of the custom schema class
      * @return A future that completes when the framework is done publishing the message
      */
-    <O> CompletableFuture<Void> publish(String topicName, O object, String serDeClassName);
+    <O> CompletableFuture<Void> publish(String topicName, O object, String schemaType);
```

Comment:
If we are enforcing schema here, why not change it to

```
<O> CompletableFuture<Void> publish(String topicName, O object, Schema<O> schema);
```

Reply:
I think we want to keep the serde concept still. Maybe I can add another publish interface with the schema. Is it ok if I do that in the following pr?

Reply:
sure

Reply:
Ya we should then mark this method as deprecated

Reply:
Serde's are not being deprecated. The only thing that is being deprecated is DefaultSerde. Thus I believe we should add another interface supporting schema rather than delete this